src/Parser/Rogue/Subtlety/Modules/ResourceTracker/ResourceTracker.js

```diff
@@ -52,7 +52,7 @@ class ResourceTracker extends Analyzer {
     this.generated += gain;  
 
     const eventResource = this.getResource(event);
-    if(eventResource !== undefined) {
+    if(!!eventResource) {
       this.current = eventResource.amount;
     }
```

Comment:
Shouldn't this just be ``if(!eventResource)``

Reply:
Actually it's checking if it *is* defined, so in this instance no exclamation mark at all is required to turn into just `if(eventResource) {`. `!!` converts a value to a boolean, but JS will do this automatically anyway. It's only really useful if you want to convert a random value (1, 0, "1", "", []) into a boolean without inverting its value.